packages/@react-spectrum/datepicker/src/DatePicker.tsx

```diff
@@ -97,13 +95,28 @@ export function DatePicker<T extends DateValue>(props: SpectrumDatePickerProps<T
   let timeGranularity = props.granularity === 'hour' || props.granularity === 'minute' || props.granularity === 'second' || props.granularity === 'millisecond' ? props.granularity : null;
   let showTimeField = (v && 'hour' in v) || !!timeGranularity;
```

Comment:
A interesting case (and perhaps a non issue):
1. Go to https://reactspectrum.blob.core.windows.net/reactspectrum/314a90c34c75615430b261fc53a935faa61a87e2/storybook/index.html?path=/story/date-and-time-datepicker--defaultvalue-zoned
2. Via keyboard, clear all the values via backspace
3. Open the popover via Option + Down Arrow
4. Note the absence of a time field despite the placeholder text having the current time because the state value has been cleared
5. Select a date.
6. Note that the time is now completely cleared from the field and there isn't a way to get it back

Perhaps a reason to only clear the state/commit the clear state upon tabbing away from the DatePicker rather than when the user deletes an entry? This is a pretty specific flow though and avoidable if the user sets a desired granularity so maybe not a big deal

Reply:
Good catch, should be fixed.